contracts/ConsensusUtils.sol

```diff
@@ -206,6 +221,26 @@ contract ConsensusUtils is EternalStorage, ValidatorSet {
     uintStorage[CURRENT_CYCLE_END_BLOCK] = block.number + getCycleDurationBlocks();
   }
 
+  function _checkJail(address[] _validatorSet) internal {
+    uint256 expectedNumberOfBlocks = CYCLE_DURATION_BLOCKS.mul(VALIDATOR_PRODUCTIVITY_BP).div(_validatorSet.length).div(10000);
+    for (uint i = 0; i < _validatorSet.length; i++) {
+      if(blockCounter(_validatorSet[i]) < expectedNumberOfBlocks) {
+        _jailVal(_validatorSet[i]);
+      } else if (getStrikes(_validatorSet[i]) != 0) {
+        _incStrikeReset(_validatorSet[i]);
+      }
+      //reset the block counter
+      _resetBlockCounter(_validatorSet[i]);
+    }
+  }
```

Comment:
can you add comments somewhere about the logic? and in general more code comments?:)
for example what is strikecount vs strikereset what do they do?